### PR TITLE
feat(bpdm): adjust bpdm service calls to new service

### PR DESCRIPTION
**Pull request by Phil91**
## Description

Change BPDM-Service Calls for the new Service
Align Service Models

## Why

structure of bpdm service has changed, therefor the models have been aligned

## Issue

N/A - Jira Ticket: CPLP-2976

## Corresponding CD PR

[#53](https://github.com/eclipse-tractusx/portal-cd/pull/53)

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/externalsystems/Bpdm.Library/BpnAccess.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    public async Task<BpdmLegalEntityDto> FetchLegalEntityByBpn(string businessPartnerNumber, string token, CancellationToken cancellationToken)
+    {
+        _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        var uri = new UriBuilder
+        {
+            Path = $"pool/api/catena/legal-entities/{Uri.EscapeDataString(businessPartnerNumber)}",
+            Query = "idType=BPN"
+        }.Uri;
+        var result = await _httpClient.GetAsync(uri.PathAndQuery, cancellationToken).ConfigureAwait(false);
+        if (!result.IsSuccessStatusCode)
```
Comment (ntruchsess) [inline src/externalsystems/Bpdm.Library/BpnAccess.cs:46]: consider use of HttpAsyncResponseMessageExtension.CatchingIntoServiceExceptionFor

File: src/externalsystems/Bpdm.Library/BpnAccess.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            throw new ServiceException($"Access to external system bpdm failed with Status Code {result.StatusCode}", result.StatusCode);
+        }
+        await using var responseStream = await result.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
+        try
+        {
+            var legalEntityResponse = await JsonSerializer.DeserializeAsync<BpdmLegalEntityDto>(
```
Comment (ntruchsess) [inline src/externalsystems/Bpdm.Library/BpnAccess.cs:53]: consider use of HttpContentJsonExtensions.ReadFromJsonAsync

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/externalsystems/Bpdm.Library/BpnAccess.cs (hunk 2 above)
Comment (Phil91) [reply src/externalsystems/Bpdm.Library/BpnAccess.cs]: changed

File: src/externalsystems/Bpdm.Library/BpnAccess.cs (hunk 1 above)
Comment (Phil91) [reply src/externalsystems/Bpdm.Library/BpnAccess.cs]: changed

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**